optuna/samplers/_nsgaiii.py

```diff
@@ -86,12 +87,16 @@ def __init__(
         swapping_prob: float = 0.5,
         seed: int | None = None,
         constraints_func: Callable[[FrozenTrial], Sequence[float]] | None = None,
+        reference_points: np.ndarray | None = None,
+        dividing_parameter: int = 3,
```

Comment:
I was going to do it in https://github.com/optuna/optuna/pull/4803 but I'll do here since it reduces the conflicts.

Reply:
Thanks for your reply. Sounds very good to make this init function keyword only. Could you apply it?

Reply:
I updated. Please check if I updated correctly as you suggested.

Reply:
As mentioned in #4803, we can accept the breaking change since `NSGAIIISampler` is still experimental. On the other hand, how about changing all the arguments like `NSGAIISampler.__init__`? It only takes keyword-only arguments.

https://github.com/optuna/optuna/blob/ddca5ac3a40f7a8934fc89d7978032006891b427/optuna/samplers/nsgaii/_sampler.py#L118-L132

Reply:
Thanks for the comments, I made all the arguments for constructor of `NSGAIIISampler` keyword-only